File: src/lib/objects/Gun.ts

export class Gun 
{
    constructor(
        public gun_size: number,
        public n_gun: number,
        public shell_weight: number,
        public range: number,
        public SRM: number,
        public ammo: number,
        public fired: boolean = true,
        ) {}
}
export function fire(gun: Gun) 
{
    gun.ammo -= 1; 
    gun.fired = true;
}
export function check_status(gun: Gun): boolean 
{
    return gun.fired;
}
export function reset(gun: Gun) 
{
    gun.fired = false;
}
    

File: public/Gun.py
from dataclasses import dataclass

@dataclass
class Gun: 
    gun_size: float
    n_gun: float
    shell_weight: float
    range: float
    SRM: float
    ammo: float
    fired: bool = True

    def fire(self): 
        self.ammo -= 1
        self.fired = True

    def check_status(self) -> bool:
        return self.fired

    def reset(self): 
        self.fired = False
